src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java

```diff
@@ -578,7 +597,12 @@ public InputStream get()
 
     private void setTime( java.util.zip.ZipEntry zipEntry, long lastModified )
     {
-        zipEntry.setTime( lastModified + ( isJava7OrLower ? 1999 : 0 ) );
+
+        if (fixedEntryModificationTime != null) {
+            zipEntry.setTime(Math.max(MIN_MS_DOS_TIME, fixedEntryModificationTime));
```

Comment:
Other values like 0L produce weird times when reading (seen something like 8. Nov 1979, which normally cannot be expressed by MS-DOS time). Not sure if we should do the round-up here.